### DISCUSSION TITLE
Search Form and Pickers for the Card Block

### DISCUSSION THREAD

**Discussion by vlakula** [Budibase/budibase#14622] [Help]
Hi!
I encountered some non-obvious logic behind the way forms work in Budibase.
As it said [here](https://docs.budibase.com/docs/adding-a-search-field) search bar (or some picker as an element with the same logic) works pretty fine with data provider when it have linked with a table or card with repeater. But the same search bar with the same setup doesn't work when I try to connect it to the card block.

For Example.
I have two duplicate screens.
The first has this structure:
![image](https://github.com/user-attachments/assets/d04413d5-4a6f-41d3-8a99-a07bc13d13f5)
The second has this structure:
![image](https://github.com/user-attachments/assets/35cd0ad6-5152-4a6e-8018-08a3013e3442)
On the second screen the search bar and pickers work fine (that's why you can see more bars and pickers - it's working). On the first the search bar just don't search anything despite the same settings of Data Provider, Form and Text Field.

Could you help me with it? Maybe I can't see something obvious. 
Thanks in advance!

**Comment by mikesealey** [comment]
Hey @vlakula, please share the settings and filters you have in the Data Provider.

**Reply by vlakula** [reply]
Of course @mikesealey! Here they are.
![image](https://github.com/user-attachments/assets/4cc66147-e522-48a9-9faa-36157a9884e8)
"Лист1" in Data parameter is a Sheet1 in my Google Sheet
![image](https://github.com/user-attachments/assets/528662e7-68d0-41e8-aa27-aee2090cc471)
"Название" ("Title" in english) is a column in Sheet1 for what I need to filter and sort. 
And the whole structure is like that:
![image](https://github.com/user-attachments/assets/54eb80b3-60f8-48e1-9359-bf18d6f3d200)
Let me explain that similar settings work on another screen, where instead of Card Block, Repeater is used.

**Reply by mikesealey** [reply]
Hey @vlakula I'm sorry for the late response.

I've tried to replicate this issue as closely as possible using GSheets as a data source, and comparing a Cards Block and a Repeater & Card combo. Unfortunately I've been unable to replicate the issue, though there have been some significant improvements since September.

Are you still able to reproduce the fault? If it's still happening on the latest version of Budibase it could be a bug, which I could investigate further with you, and pass over to the engineers to look for a solution.

**Reply by vlakula** [reply]
Hi @mikesealey!
I have already resolve this problem via repeater so I just deleted the screen that use a card block. And at the moment I can't check this logic via card block. 
I think you could close this ticket for now. If there will be the same issue I will open a new.
Thanks!
